opensrp-opd/src/main/java/org/smartregister/opd/widgets/OpdBarcodeFactory.java

```diff
@@ -0,0 +1,85 @@
+import com.vijay.jsonwizard.interfaces.JsonApi;
+import com.vijay.jsonwizard.interfaces.OnActivityResultListener;
+import com.vijay.jsonwizard.widgets.BarcodeFactory;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.smartregister.opd.listener.LookUpTextWatcher;
+import org.smartregister.opd.utils.OpdConstants;
+
+import java.util.List;
+
+import timber.log.Timber;
+
+import static android.app.Activity.RESULT_OK;
+
+public class OpdBarcodeFactory extends BarcodeFactory {
+
+    private JsonFormFragment jsonFormFragment;
+    private boolean forLookUp;
+
+    @Override
+    public List<View> getViewsFromJson(String stepName, Context context, JsonFormFragment formFragment, JSONObject jsonObject, CommonListener listener) throws Exception {
+        List<View> viewList = super.getViewsFromJson(stepName, context, formFragment, jsonObject, listener);
+        this.jsonFormFragment = formFragment;
+        try {
+            this.forLookUp = jsonObject.has(OpdConstants.KEY.LOOK_UP) &&
+                    jsonObject.get(OpdConstants.KEY.LOOK_UP).toString().equalsIgnoreCase(Boolean.TRUE.toString());
+        } catch (JSONException e) {
+            Timber.e(e);
+            e.printStackTrace();
+        }
+        return viewList;
+    }
+
+    @Override
+    public List<View> getViewsFromJson(String stepName, Context context, JsonFormFragment formFragment, JSONObject jsonObject, CommonListener listener, boolean popup) {
+        List<View> viewList = super.getViewsFromJson(stepName, context, formFragment, jsonObject, listener, popup);
+        this.jsonFormFragment = formFragment;
+        try {
+            this.forLookUp = jsonObject.has(OpdConstants.KEY.LOOK_UP) &&
+                    jsonObject.get(OpdConstants.KEY.LOOK_UP).toString().equalsIgnoreCase(Boolean.TRUE.toString());
+        } catch (JSONException e) {
+            Timber.e(e);
+            e.printStackTrace();
+        }
+        return viewList;
+    }
+
+    @Override
+    protected void addOnBarCodeResultListeners(Context context, final MaterialEditText editText) {
+        if (context instanceof JsonApi) {
+            JsonApi jsonApi = (JsonApi) context;
+            jsonApi.addOnActivityResultListener(JsonFormConstants.BARCODE_CONSTANTS.BARCODE_REQUEST_CODE,
+                    new OnActivityResultListener() {
+                        @Override
+                        public void onActivityResult(int requestCode,
+                                                     int resultCode, Intent data) {
+                            if (requestCode == JsonFormConstants.BARCODE_CONSTANTS.BARCODE_REQUEST_CODE && resultCode == RESULT_OK) {
+                                if (data != null) {
+                                    Barcode barcode = data.getParcelableExtra(JsonFormConstants.BARCODE_CONSTANTS.BARCODE_KEY);
+                                    Timber.d("Scanned QR Code %s ", barcode.displayValue);
+                                    if( forLookUp ) {
```

Comment:
- Reformat the file 
- add the correct nullable & nonNull annotation
- Avoid static imports - `android.app.Activity.RESULT_OK` Static imports are not easy to locate their origin

Reply:
done